src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -266,7 +266,12 @@ private static bool IsByteArray(string fileRefType)
             return fileRefType.IndexOf("System.Byte[]") != -1 && fileRefType.IndexOf("mscorlib") != -1;
         }
 
-        private static bool IsMemoryStream(string fileRefType)
+        internal static bool IsString(string fileRefType)
+        {
+            return fileRefType == StringTypeName;
+        }
+
+        internal static bool IsMemoryStream(string fileRefType)
         {
             return fileRefType.Equals(MemoryStreamTypeNameDesktopFramework, StringComparison.Ordinal);
```

Comment:
Does this only work when building on the Desktop framework?  Or should `MemoryStreamTypeNameDesktopFramework` be renamed?

Reply:
This is very confusing. This codepath works on both core and desktop, but it only supports type names _serialized into the resx_ that match the desktop names. Other tooling is responsible for ensuring that those names are used at resx-writing time. Some further discussion at https://github.com/microsoft/msbuild/pull/4607#discussion_r315426184